lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -555,6 +556,22 @@ instance FromJSON ApiStakePoolMetrics where
 instance ToJSON ApiStakePoolMetrics where
     toJSON = genericToJSON defaultRecordTypeOptions
 
+instance FromJSON (ApiT StakePoolMetadata) where
+    parseJSON = withObject "StakePoolMetadata" $ \v -> ApiT <$>
+        (StakePoolMetadata
+            <$> (v .: "ticker" >>= parseTicker)
+            <*> (v .: "homepage")
+            <*> (v .: "pledge_address"))
+      where
+        parseTicker = eitherToParser . left ShowFmt . fromText
+
+instance ToJSON (ApiT StakePoolMetadata) where
+    toJSON (ApiT (StakePoolMetadata t h pa)) = object
+        [ "ticker" .= toText t
+        , "homepage" .= h
+        , "pledge_address" .= pa
+        ]
```

Comment:
**[suggestion: maintainability]** 

Use `Generic` to derive JSON instances. 

**Why**
1. Makes for more maintainable code in the long-run (avoid repetition of names and, make sure that the JSON encoding are consistent across many types).
2. More consistent with the style used across the rest of the module.

Reply:
OK done.